src/main/java/de/thetaphi/forbiddenapis/maven/AbstractCheckMojo.java

```diff
@@ -237,18 +242,15 @@ public abstract class AbstractCheckMojo extends AbstractMojo implements Constant
   /** The project packaging (pom, jar, etc.). */
   @Parameter(defaultValue = "${project.packaging}", readonly = true, required = true)
   private String packaging;
-  
-  @Component
-  private ArtifactFactory artifactFactory;
 
   @Component
-  private ArtifactResolver artifactResolver;
-  
-  @Parameter(defaultValue = "${project.remoteArtifactRepositories}", readonly = true, required = true)
-  private List<ArtifactRepository> remoteRepositories;
+  private RepositorySystem repoSystem;
 
-  @Parameter(defaultValue = "${localRepository}", readonly = true, required = true)
-  private ArtifactRepository localRepository;
+  @Parameter(defaultValue = "${repositorySystemSession}", readonly = true, required = true)
+  private RepositorySystemSession repoSession;
+
+  @Parameter(defaultValue = "${project.remoteProjectRepositories}", readonly = true, required = true)
```

Comment:
I think that `${project.remotePluginRepositories}` would be semantically correct, but you always used the project repository to resolve signature artifacts.